src/components/plots/plot.ts

```diff
@@ -163,30 +173,29 @@ module Plottable {
      */
     public project(attrToSet: string, accessor: any, scale?: Scale<any, any>) {
       attrToSet = attrToSet.toLowerCase();
-      var previousProjection = this._attrBindings.get(attrToSet);
-      var previousScale = previousProjection && previousProjection.scale;
-
       accessor = Utils.Methods.accessorize(accessor);
-      this._attrBindings.set(attrToSet, {accessor: accessor, scale: scale, attribute: attrToSet});
-      this._updateExtentsForAttr(attrToSet);
-
-      if (previousScale) {
-        if (this._scales().indexOf(previousScale) !== -1) {
-          previousScale.offUpdate(this._renderCallback);
-          previousScale.removeExtentProvider(this._extentProvider);
-        }
-        previousScale._autoDomainIfAutomaticMode();
-      }
-
-      if (scale) {
-        scale.onUpdate(this._renderCallback);
-        scale.addExtentProvider(this._extentProvider);
-        scale._autoDomainIfAutomaticMode();
-      }
+      this._setupAttr(attrToSet, accessor, scale);
       this.render(); // queue a re-render upon changing projector
       return this;
     }
 
+    protected _setupProperty(property: string, value: any, scale: Scale<any, any>) {
```

Comment:
not sure "setup" is the right verb here. Since we're setting a binding, maybe "`_bindProperty()`"?